python/cuml/prims/stats/covariance.py

```diff
@@ -156,3 +196,94 @@ def cov(x, y, mean_x=None, mean_y=None, return_gram=False, return_mean=False):
         return cov_result, mean_x, mean_y
     elif return_gram and return_mean:
         return cov_result, gram_matrix, mean_x, mean_y
+
+
+@cuml.internals.api_return_any()
+def _cov_sparse(x, return_gram=False, return_mean=False):
+    """
+    Computes the mean and the covariance of matrix X of
+    the form Cov(X, X) = E(XX) - E(X)E(X)
+
+    This is a temporary fix for issue #5475, where the
+    operation `x.T.dot(x)` did not work for larger
+    sparse matrices.
```

Comment:
Sorry, I didn't make that clear enough earlier. The issue that motivates and should eventually trigger the removal of this workaround would be a fix of https://github.com/cupy/cupy/issues/7699 .